datastore/longhorn.go

```diff
@@ -233,6 +232,59 @@ func (s *DataStore) ListSettings() (map[types.SettingName]*longhorn.Setting, err
 	return itemMap, nil
 }
 
+// AnnotateAWSIAMRoleArn ensures that the running pods of the manager as well as the replica instance managers.
+// have the correct AWS IAM role arn assigned to them based on the passed `awsIAMRoleArn`
+func (s *DataStore) AnnotateAWSIAMRoleArn(controllerID, awsIAMRoleArn string) (bool, error) {
+	update := false
+
+	selector, err := s.getManagerSelector()
+	if err != nil {
+		return update, err
+	}
+	managerPods, err := s.pLister.Pods(s.namespace).List(selector)
+	if err != nil {
+		return update, err
+	}
+
+	selector, err = metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: types.GetInstanceManagerLabels("", "", types.InstanceManagerTypeReplica),
+	})
+	if err != nil {
+		return update, err
+	}
+	imPods, err := s.pLister.Pods(s.namespace).List(selector)
```

Comment:
For AWS IAM Role, we only can apply to the `pod` level instead of `process` level, so that means this will be a problem in the future for mulitple AWS credentials support.

But, I would say this is the limitation when using `assume role`, so it can be documented when we support multiple AWS credentials in the future.